frontend/src/scenes/session-recordings/player/SessionRecordingPlayer.tsx

```diff
@@ -168,6 +171,9 @@ export function SessionRecordingPlayer(props: SessionRecordingPlayerProps): JSX.
     const layoutStacking = compactLayout ? InspectorStacking.Vertical : preferredInspectorStacking
     const isVerticallyStacked = layoutStacking === InspectorStacking.Vertical
 
+    const lessThanFiveMinutesOld = dayjs().diff(start, 'minute') <= 5
```

Comment:
This is only helpful if lagging is < 5 minutes right?

Reply:
I'd lean toward using a more generic message instead of "It should be ready to watch in a few minutes" and do not use a magic number check, otherwise this UI message won't appear and playing recordings will still not be great when lagging is > 5 min.

> It should be ready to watch once all the recording is fully ingested

So we don't give false hopes whenever the recording is broken/missing data.

Reply:
How would you expect it to appear to a user in cases where we've entirely missed the full snapshot? Without the time component (which ideally would include the expected ingestion lag) we would have no hook to change the state and the recording would never become "ready".

The time constraint is a little arbitrary to be fair but this is more of a catch all in cases where the recording is relatively new & doesn't yet have a full snapshot so we're basically assuming it is unplayable.

Reply:
> How would you expect it to appear to a user in cases where we've entirely missed the full snapshot? Without the time component (which ideally would include the expected ingestion lag) we would have no hook to change the state and the recording would never become "ready".

Good point, but this is only if the person does not refresh the page and stays in the same recording for some time, right? I wonder if this is ever the case, usually users would just come to the next recording unless they want to see that one specifically.

My point was more about the use case where we missed the full snapshot for any reason and likely won't ever be playable but we're promising this is gonna be ready within a few minutes.

Happy to move this forward and improve if needed, was just sharing my thoughts since the lag is constantly changing and we're often debugging recordings with missing full snapshots and sometimes they are even from hours ago.

Reply:
> Good point, but this is only if the person does not refresh the page and stays in the same recording for some time, right?
It actually triggers when the recording is visited and does not perform any kind of polling for new data. So what I'd expect to see happening most often is that someone gets this screen and just moves on to the next recording. If they come back to it in an hour it will either have the full snapshot or now be unplayable. Worth noting there's also a reload button on the banner that users can click that will reload the recording (which may or may not have a recording at that point)

> My point was more about the use case where we missed the full snapshot for any reason and likely won't ever be playable but we're promising this is gonna be ready within a few minutes.
Yeah, that is a problem. The problem is we just don't know if it's coming. It could be that it's never coming or it could be that the full snapshot just hasn't been ingested yet.

This certainly isn't the optimal solution. My hope is that it solves 90% of the cases where someone opens a support ticket because they cannot playback a recording in the last 10 minutes. In the majority of those cases the full snapshot is on it's way, the cases where we entirely miss the snapshot are much rarer. Ultimately I hope we can remove this screen once we figure out _why_ the snapshots are being dropped in the first place